crates/image_viewer/src/image_viewer.rs

```diff
@@ -145,6 +160,35 @@ impl Item for ImageView {
     }
 }
 
+fn image_info(
+    image: &ImageItem,
+    project: &Project,
+    cx: &AppContext,
+) -> Result<(u32, u32, u64), String> {
+    let worktree = project
+        .worktree_for_id(image.project_path(cx).worktree_id, cx)
+        .ok_or_else(|| "Could not find worktree for image".to_string())?;
+    let worktree_root = worktree.read(cx).abs_path();
+
+    let path = if image.path().is_absolute() {
+        image.path().to_path_buf()
+    } else {
+        worktree_root.join(image.path())
+    };
+
+    if !path.exists() {
+        return Err(format!("File does not exist at path: {:?}", path));
+    }
+
+    let img = image::open(&path).map_err(|e| format!("Failed to open image: {}", e))?;
```

Comment:
This and the `metadata()` call below should not be here. We don't want to block the UI thread with filesystem access. This whole function should move to `ImageItem::try_open` and `ImageItem::reload`